library/std/src/sys/windows/path.rs

```diff
@@ -263,9 +306,21 @@ pub(crate) fn maybe_verbatim(path: &Path) -> io::Result<Vec<u16>> {
 
 /// Make a Windows path absolute.
 pub(crate) fn absolute(path: &Path) -> io::Result<PathBuf> {
-    if path.as_os_str().bytes().starts_with(br"\\?\") {
-        return Ok(path.into());
+    let path = path.as_os_str();
+    // NULs in verbatim paths are rejected for consistency.
+    if path.bytes().contains(&0) {
+        return Err(io::const_io_error!(
+            io::ErrorKind::InvalidInput,
+            "strings passed to WinAPI cannot contain NULs",
+        ));
     }
```

Comment:
Could this only be done if the prefix is actually verbatim? Otherwise this check is essentially done twice in the non-verbatim case. It probably doesn't matter too much though.

Reply:
I'm conflicted about this. The newest implementation is likely the most performant, but maybe it would make more sense to always call `to_u16s` to avoid code duplication.